iox_query_influxrpc/src/lib.rs

```diff
@@ -246,51 +246,13 @@ impl InfluxRpcPlanner {
 
-        let metadata_ctx = ctx.child_ctx("apply_predicate_to_metadata");
-        let metadata_ctx = &metadata_ctx; // needed to use inside the move closure
-
         let table_predicates = rpc_predicate
             .table_predicates(namespace.as_meta())
             .context(CreatingPredicatesSnafu)?;
         let tables: Vec<_> =
             table_chunk_stream(Arc::clone(&namespace), false, &table_predicates, &ctx)
-                .try_filter_map(|(table_name, predicate, chunks)| async move {
-                    // Identify which chunks can answer from its metadata and then record its table,
-                    // and which chunks needs full plan and group them into their table
-                    let mut chunks_full = vec![];
-
-                    for chunk in cheap_chunk_first(chunks) {
-                        trace!(chunk_id=%chunk.id(), %table_name, "Considering table");
-
-                        // If the chunk has delete predicates, we need to scan (do full plan) the data to eliminate
-                        // deleted data before we can determine if its table participates in the requested predicate.
-                        if chunk.has_delete_predicates() {
-                            chunks_full.push(chunk);
-                        } else {
-                            // Try and apply the predicate using only metadata
-                            let pred_result = chunk
-                                .apply_predicate_to_metadata(metadata_ctx, predicate)
-                                .context(CheckingChunkPredicateSnafu {
-                                    chunk_id: chunk.id(),
-                                })?;
-
-                            match pred_result {
-                                PredicateMatch::AtLeastOneNonNullField => {
-                                    trace!("Metadata predicate: table matches");
-                                    // Meta data of the table covers predicates of the request
-                                    return Ok(Some((table_name, None)));
-                                }
-                                PredicateMatch::Unknown => {
-                                    trace!("Metadata predicate: unknown match");
-                                    // We cannot match the predicate to get answer from meta data, let do full plan
-                                    chunks_full.push(chunk);
-                                }
-                                PredicateMatch::Zero => {
-                                    trace!("Metadata predicate: zero rows match");
-                                } // this chunk's table does not participate in the request
-                            }
-                        }
-                    }
+                .try_filter_map(|(table_name, table_schema, predicate, chunks)| async move {
+                    let chunks_full = prune_chunks(&table_schema, chunks, predicate);
```

Comment:
I got confused at first about why this call to `prune_chunks` was doing something different than the one right below it. Then I saw that this one was applying a table predicate and the one below was applying the entire predicate

I think the code would be easier to understand if the binding to `table_predicates` in the closure was `table_predicate` rather than `predicate`:

```suggestion
                    let chunks_full = prune_chunks(&table_schema, chunks, table_predicate);
```

But that is entirely a style suggestion. The same comment applies to the other functions below.

Reply:
done